extensions/amp-story/1.0/amp-story-page.js

```diff
@@ -683,6 +681,36 @@ export class AmpStoryPage extends AMP.BaseElement {
     return Promise.all(mediaPromises).then(() => this.markPageAsLoaded_());
   }
 
+  /**
+   * @return {!Promise}
+   * @private
+   */
+  waitForPlaybackMediaLayout_() {
+    const mediaSet = toArray(
+      this.getMediaBySelector_(Selectors.ALL_PLAYBACK_AMP_MEDIA)
+    );
+
+    const mediaPromises = mediaSet.map((mediaEl) => {
+      return new Promise((resolve) => {
+        switch (mediaEl.tagName.toLowerCase()) {
+          case 'audio': // Already laid out.
```

Comment:
IIUC, `audio` is not a possible `tagName` from the set queried from `ALL_PLAYBACK_AMP_MEDIA`, but we still check here because it may be returned by `getMediaBySelector_` when elements are found via fie, which queries `ALL_IFRAMED_MEDIA`. If that is the case why isn't `video` also an option then?

Reply:
I'm confused about the comment `// Already laid out.` -- I know `amp-audio` and `amp-video` generate their underlying `audio` and `video` elements, but aren't they nested and therefore not accessible here? Is this referring to something else?

Reply:
Final comment on this piece -- If you do want to special case `audio` for being laid out, WDYT about moving it down before the `default` case so it can also fall through?

Reply:
Thanks for your comments!
A bit more context to explain the selector having `audio` when it's actually not really needed: I wanted to write a selector that retrieves all the parents (`amp-*`, not the actual `HTMLMediaElement`) of the elements that will eventually have to be registered in the media pool, so we can wait for them to be laid out and avoid race conditions where the inner `video` or `audio` doesn't exist yet.
To come back to your actual point: we'd like to always get the `amp-audio` and `amp-video` and wait for them to be laid out, but when a `story-page` has a `background-audio="foo.mp3"` attribute, we create an `audio` element without a `amp-audio` wrapper, so I wanted to take that into account. 

Happy to remove the `audio` if you think that's too confusing, I added it for consistency with the selector name. Maybe we could rename the selector from `ALL_PLAYBACK_AMP_MEDIA` to something else.

Reply:
The autogenerated `audio` element without an `amp-audio` wrapper would not get selected by the set in `ALL_PLAYBACK_AMP_MEDIA`, right? Does this code path only get executed when `background-audio` is set _and_ it queries from `ALL_IFRAMED_MEDIA`? Thanks for your patience improving my amp-story literacy here 😅

Reply:
> The autogenerated audio element without an amp-audio wrapper would not get selected by the set in ALL_PLAYBACK_AMP_MEDIA, right?

Actually it does, but you're right that does not make much sense. I'll send a follow up PR either renaming the selectors or changing their content. Thank you for bringing this up! :))